deep-map/index.d.ts

```diff
@@ -78,4 +85,6 @@ export type DeepMapStore<T extends BaseDeepMap> = {
  * @param init Initialize store and return store destructor.
  * @returns The store object with methods to subscribe.
  */
-export function deepMap<T extends BaseDeepMap>(init?: T): DeepMapStore<T>
+export function deepMap<Value extends BaseDeepMap | undefined>(
+  value?: Value
```

Comment:
We need to allow miss `value` only if `Value` type allows `undefined`

Reply:
Yes, that's the part I couldn't figure out. Not sure if it's the function overloading syntax I'm not getting right or if it requires some other sort of type wrangling.

Reply:
We can use two definitions like:

```ts
export function map<Value extends object>(value: Value): PreinitializedMapStore<Value>
export function map<Value extends undefined>(value?: Value): PreinitializedMapStore<Value>
```

Reply:
I feel that was exactly what I tried, but I'll have another look.

Reply:
Try to change the order, maybe we need to put `undefined` first

Reply:
No, doesn't matter. Neither of them works. Also tried having then, and a third one that covers both cases. 

Not having one that covers both makes both versions complain:

`Type '{ a: number; }' does not satisfy the constraint 'undefined'.`
`Type 'Record<string, unknown> | undefined' does not satisfy the constraint 'object'.`

And adding the overload that covers all three then swallows the cases that should be errors. Not sure if it's a restriction in having union overloads with the same number of parameters (Similar to https://github.com/microsoft/TypeScript/issues/14107)